documentation/general/dash-sonic-hld.md

```diff
@@ -386,13 +417,59 @@ key                      = DASH_VNET_MAPPING_TABLE:vnet:ip_address ; CA-PA mappi
 action_type              = routing_type              ; reference to routing type
 underlay_ip              = ip_address                ; PA address for the CA
 mac_address              = MAC address as string     ; Inner dst mac
-metering_bucket          = bucket_id                 ; metering and counter
+metering_bucket          = bucket_id                 ; metering bucket id
+override_meter           = bool                      ; override the metering bucket coming from the route table
 use_dst_vni              = bool                      ; if true, use the destination VNET VNI for encap. If false or not specified, use source VNET's VNI
 overlay_sip              = ip_address                ; overlay src ip if routing_type is {privatelink}, transform last 32 bits from packet 
 overlay_dip              = ip_address                ; overlay dst ip if routing_type is {privatelink} 
 ```
 
-### 3.2.9 DASH orchagent (Overlay)
+### 3.2.9 METER
+
+```
+DASH_METER_POLICY:{{meter_policy_id}} 
+    "ip_version": {{ipv4/ipv6}}
+```
+
+```
+key                      = DASH_METER_POLICY:meter_policy_id ; policy_id as key
+; field                  = value 
+ip_version               = IP version (IPv4/IPv6)
+```
+
+
+```
+DASH_METER_RULE:{{meter_policy_id}}:{{rule_num}}
+    "priority": {{priority}}
+    "ip_prefix":{{ip_prefix}}
+    "metering_bucket": {{bucket_id}}
+```
+
+```
+key                      = DASH_METER_RULE:meter_policy_id:rule_num ; unique rule num within the policy.
+; field                  = value 
+priority                 = INT32 value  ; priority of the rule: lower the value, higher the priority
+ip_prefix                = ip_prefix    ; ip prefix for matching
+metering_bucket          = bucket_id    ; metering bucket id
+```
+
+
+```
+DASH_METER:{{eni}}:{{bucket_id}}
```

Comment:
I take it that the DASH_METER is essentially the "bucket" which we refer to in this document.  Also, it has a "bucket_id".  Why do we need "ENI" as a key here?  Since, as you mentioned earlier, the bucket IDs are globally unique across all VNETs in a region, I don't believe we need the ENI as a key here.  Also, bucket IDs are assigned to policy meters and policy meters are subsequently attached to ENIs, the usage of ENI here doesn't make sense.  Am I missing something here?  Thanks!

Reply:
The billing is on a per ENI basis, so the global metering bucket id needs to be associated with each ENI where metering is required. Also metering buckets are associated with mappings as well which is per VNET